openrgb: 0.8 -> 0.9

panda2134 opened the pull request:
## Description of changes

https://gitlab.com/CalcProgrammer1/OpenRGB/-/tags/release_0.9

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

panda2134 committed:
openrgb: 0.8 -> 0.9

Tungsten842 commented:
I think that you should also update: [openrgb-plugin-effects](https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=openrgb) and [openrgb-plugin-hardwaresync](https://search.nixos.org/packages?channel=unstable&show=openrgb-plugin-hardwaresync&from=0&size=50&sort=relevance&type=packages&query=openrgb)

panda2134 committed:
openrgb-plugin-effects: 0.8 -> 0.9

removed patch 75f1b3617d9cabfb3b04a7afc75ce0c1b8514bc0, as this commit
has been included in v0.9.

panda2134 committed:
openrgb-plugin-hardwaresync: 0.8 -> 0.9

removed the following patches as they are already present in the current
version:

df2869d679ea43119fb9b174cd0b2cb152022685
bfbaa0a32ed05112e0cc8b6b2a8229945596e522

panda2134 commented:
> I think that you should also update: [openrgb-plugin-effects](https://gitlab.com/OpenRGBDevelopers/OpenRGBEffectsPlugin/-/releases) and [openrgb-plugin-hardwaresync](https://gitlab.com/OpenRGBDevelopers/OpenRGBHardwareSyncPlugin/-/releases)

Updated with [3727072](https://github.com/NixOS/nixpkgs/pull/250358/commits/3727072bdfb2d416a0637c2f5b96a6fdb04209a5) and [54efceb](https://github.com/NixOS/nixpkgs/pull/250358/commits/54efcebf9b65f6b6f53910ef55e897f48083e9d7)

khaneliman commented:
Do I need to do anything else in particular to test this package? I get ```Cannot mix incompatible Qt library (5.15.9) with this library (5.15.10)``` at runtime.

panda2134 commented:
> Do I need to do anything else in particular to test this package? I get `Cannot mix incompatible Qt library (5.15.9) with this library (5.15.10)` at runtime.

Did you see this when running `openrgb`? I failed to reproduce that.

khaneliman commented:
> > Do I need to do anything else in particular to test this package? I get `Cannot mix incompatible Qt library (5.15.9) with this library (5.15.10)` at runtime.
> 
> Did you see this when running `openrgb`? I failed to reproduce that.

Yeah, just did a nixpkgs-review pr 250358 and ./results/openrgb/bin/openrgb

khaneliman commented:
Result of `nixpkgs-review pr 250358` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>openrgb</li>
    <li>openrgb-plugin-effects</li>
    <li>openrgb-plugin-hardwaresync</li>
    <li>openrgb-with-all-plugins</li>
  </ul>
</details>

panda2134 commented:
> EDIT: well i get the same result when building from master so just ignore my runtime result executing the binary directly, i guess.

Yeah that could be the reason. I'm testing with `nixpkgs-unstable` which is a bit newer than `nixos-unstable`, so the Qt version might differ.

fgaz commented on pkgs/applications/misc/openrgb-plugins/hardwaresync/default.nix:
```diff
@@ -14,28 +14,15 @@
   };
 
-  patches = [
-    (fetchpatch {
```
`fetchpatch` is unused now, remove it from the inputs

fgaz reviewed (changes requested):
Please squash 9dc89cabff69f37572748392dc39afc64a93c5ef into 54efcebf9b65f6b6f53910ef55e897f48083e9d7

fgaz commented:
> Do I need to do anything else in particular to test this package?
> 
> ```
> Cannot mix incompatible Qt library (5.15.9) with this library (5.15.10)
> Aborted (core dumped)
> ```

Works for me on NixOS 23.05

SuperSamus commented:
#242899 needs to be closed.
